src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/IModelExpressionProvider.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq.Expressions;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+
+namespace Microsoft.AspNetCore.Mvc.Rendering
+{
+    public interface IModelExpressionProvider
```

Comment:
Make this interface generic so that the `RazorPage<TModel>` property can be strongly typed i.e. `IModelExpresionProvider<TModel>`. Inject the property w/ the correct `TModel`.